lib/hostasyn.c

```diff
@@ -147,6 +147,15 @@ Curl_addrinfo *Curl_getaddrinfo(struct connectdata *conn,
                                 int port,
                                 int *waitp)
 {
+  struct Curl_easy *data = conn->data;
+
+  /* Call the resolver start callback, if defined */
+  if(data->set.resolver_callbacks.resolver_start_cb) {
+    data->set.resolver_callbacks.
```

Comment:
so here I would argue there should be a return code and handling of that, as mentioned above

Reply:
Sure, thanks for the suggestion - I've fixed the failing TCs and added the example, will add the two other issues you mention (recursive CB attempt + return code) on a subsequent commit. Thanks for the help.

Reply:
Just submitted code with the requested changes, updated TCs and doc.
Thanks,